src/test/groovy/graphql/normalized/ExecutableNormalizedOperationToAstCompilerTest.groovy

```diff
@@ -2153,26 +2153,129 @@ abstract class ExecutableNormalizedOperationToAstCompilerTest extends Specificat
                  v3: [arg1: "barFragArg"],
                  v4: [arg1: "barArg"],
                  v5: [arg1: "fooArg"]]
+
+        //
+        // Test the opposite - when we use no variables predicate everything should be inlined
+        //
+        when: "it has no variables"
+
+        fields = createNormalizedFields(schema, query, [v0: [1, 2, 3],
+                                                        v1: [[arg1: "v1", arg2: [[arg1: "v1.1"]]], [arg1: "v2"], [arg1: "v3"]],
+                                                        v2: [[arg1: "fooNonNullArg1"], [arg1: "fooNonNullArg2"]],
+                                                        v3: [arg1: "barFragArg"],
+                                                        v4: [arg1: "barArg"],
+                                                        v5: [arg1: "fooArg"]])
+
+        result = localCompileToDocument(schema, QUERY, "named", fields, noVariables)
+        document = result.document
+        vars = result.variables
+        ast = printDoc(document)
+
+        then: "they should be inlined as values"
+
+        // no vars created
+        vars == [:]
+
+        // everything inlined
+        ast == '''query named {
+  foo(arg: {arg1 : "fooArg"}) {
+    bar(arg: {arg1 : "barArg"}) {
+      baz {
+        ... on ABaz {
+          a
+          boo(arg: {arg1 : "barFragArg"})
+        }
+      }
+    }
+  }
+  fooNonNull(arg: [{arg1 : "fooNonNullArg1"}, {arg1 : "fooNonNullArg2"}])
+  listField1(arg: [1, 2, 3])
+  listField2(arg: [{arg1 : "v1", arg2 : [{arg1 : "v1.1"}]}, {arg1 : "v2"}, {arg1 : "v3"}])
+}
+'''
     }
 
-    private ExecutableNormalizedOperation createNormalizedTree(GraphQLSchema schema, String query, Map<String, Object> variables = [:]) {
+    def "can extract variables or inline values for directives on the query"() {
```

Comment:
Added this test for directives - it does NOT work - directives are ignored and removed